misc/dbt-materialize/dbt/include/materialize/macros/adapters.sql

```diff
@@ -86,6 +86,40 @@
     {{ sql }}
 {%- endmacro %}
 
+{% macro materialize__create_source(relation, sql) -%}
+  {% set contract_config = config.get('contract') %}
+  {% if contract_config.enforced %}
+    {{exceptions.warn("Model contracts cannot be enforced for custom materializations (see dbt-core #7213)")}}
+  {%- endif %}
+
+  {%- set cluster = config.get('cluster', target.cluster) -%}
+
+  create source {{ relation }}
+  {% if cluster %}
+    in cluster {{ cluster }}
+  {% endif %}
```

Comment:
I was thinking that we probably want to include the `FROM` here too, for consistency with `CREATE VIEW ... AS`. I.e., the preposition is stripped off in both cases.

```suggestion
  {% endif %}
from
```